packages/sanity/src/core/form/contexts/GetFormValue.tsx

```diff
@@ -0,0 +1,60 @@
+import React, {createContext, ReactNode, useCallback, useContext, useMemo, useRef} from 'react'
+import {Path} from '@sanity/types'
+import {getValueAtPath} from '../../field'
+import type {FormDocumentValue} from '../types'
+
+/**
+ * @internal
+ * @hidden
+ */
+export type GetFormValueContextValue = (path: Path) => unknown
+
+const GetFormValueContext = createContext<GetFormValueContextValue | null>(null)
```

Comment:
No blocker for merge, but I think it is nice to split the provider, context, and hook(s) into different files, mainly because that makes it a bit easier and faster to find things in the codebase by utilizing the file search in the editor (at least in VSCode)!

Reply:
Great point! Should be better now after db64732b74ba3b28f3984f9d8c8c99cb40921fbb

Reply:
Actually, looks like db64732b74ba3b28f3984f9d8c8c99cb40921fbb needs some more work, so I'm reverting that. Doesn't feel critical, can address it later :)